### PR TITLE
03 java services

### PR DESCRIPTION
**Pull request by m1ks-wq**

### CODE REVIEW

File: 03-JavaServiceSkeleton/SetterApp/app/src/main/java/com/example/setterapp/MainActivity.java [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+        setContentView(R.layout.activity_main);
+
+        final EditText edtX = (EditText) findViewById(R.id.edtX);
+        final Button btnADD = (Button) findViewById(R.id.btnAdd);
+        btnADD.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                String sx = edtX.getText().toString();
+                try {
+                    int x = Integer.valueOf(sx);
```
Comment (BulldozerBSG) [inline 03-JavaServiceSkeleton/SetterApp/app/src/main/java/com/example/setterapp/MainActivity.java:49]: What happens if sx contains no number?

File: 03-JavaServiceSkeleton/SetterApp/app/src/main/java/com/example/setterapp/MainActivity.java [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+                    myService.add(x);
```
Comment (BulldozerBSG) [inline 03-JavaServiceSkeleton/SetterApp/app/src/main/java/com/example/setterapp/MainActivity.java:50]: What happens if mService contains null?

File: 03-JavaServiceSkeleton/GetterApp/app/src/main/java/com/example/getterapp/MainActivity.java [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_main);
+
+        final EditText edtX = (EditText) findViewById(R.id.edtX);
+        final Button btnGET = (Button) findViewById(R.id.btnGet);
+        btnGET.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                try {
+                    edtX.setText(String.valueOf(myService.get()));
```
Comment (BulldozerBSG) [inline 03-JavaServiceSkeleton/GetterApp/app/src/main/java/com/example/getterapp/MainActivity.java:48]: What happens if mService contains null?

**Review by BulldozerBSG — CHANGES_REQUESTED** [review]

File: 03-JavaServiceSkeleton/SetterApp/app/src/main/java/com/example/setterapp/MainActivity.java (hunk 1 above)
Comment (m1ks-wq) [reply 03-JavaServiceSkeleton/SetterApp/app/src/main/java/com/example/setterapp/MainActivity.java:49]: Fixed.

File: 03-JavaServiceSkeleton/GetterApp/app/src/main/java/com/example/getterapp/MainActivity.java (hunk 3 above)
Comment (m1ks-wq) [reply 03-JavaServiceSkeleton/GetterApp/app/src/main/java/com/example/getterapp/MainActivity.java:48]: Fixed.

File: 03-JavaServiceSkeleton/SetterApp/app/src/main/java/com/example/setterapp/MainActivity.java (hunk 2 above)
Comment (m1ks-wq) [reply 03-JavaServiceSkeleton/SetterApp/app/src/main/java/com/example/setterapp/MainActivity.java:50]: Fixed.

**Review by BulldozerBSG — APPROVED** [review]

**Pull request merged**